### ISSUE TITLE
OpenGL 3D rendering support

### ISSUE THREAD

**Issue by ToniBC** [libretro/scummvm#55]
In the standalone emulator, we have the option of using OpenGL, which allows us in several 3D games to increase the resolution or use the 3D acceleration options, freeing up the CPU load and improving the game. Like Grim Fandango and others.

It would be nice to be able to include that in the Libretro version.

Thank you.

**Comment by spleen1981** [comment]
OpenGL support (2D) has now been added 42dc7b31f4d30cc56a06d9b64d0f479d62e3ae2b

**Comment by rtissera** [comment]
@spleen1981 great addition, can it be upstreamed ?

**Comment by KamiLordus** [comment]
@spleen1981 thanks for adding hardware acceleration.
Is it possible for The Longest Journey to also use hardware acceleration? Then the characters and text look smoothed (antyaliasing?)
Currently, the game does not use hardware acceleration, and there are graphical problems in the first scene with April. Colorful pixels around the characters, which is difficult to show in a screenshot.
![Longest Journey, The (CD Windows)-240418-001350](https://github.com/libretro/scummvm/assets/123302593/ab1ffb57-fed2-425b-9027-92796f280a16)

**Comment by spleen1981** [comment]
> @spleen1981 great addition, can it be upstreamed ?

not sure what do you mean, however the updated version should be already available from retroarch online updater and in sync with upstream `scummvm/scummvm` repo

**Comment by spleen1981** [comment]
@KamiLordus that would require specific OpenGL 3D support, not available yet in the core.

**Comment by rtissera** [comment]
> > @spleen1981 great addition, can it be upstreamed ?
> 
> not sure what do you mean, however the updated version should be already available from retroarch online updater and in sync with upstream `scummvm/scummvm` repo

Yes, I mean send it to scummvm/scummvm repo as a pull request

**Comment by spleen1981** [comment]
> Yes, I mean send it to scummvm/scummvm repo as a pull request 

Yes,  actually all LIBRETRO commits in master are pushed upsteam first and then synced here

**Comment by spleen1981** [comment]
openGL 3D rendering support is now implemented @ 55fcf4050ac1102638cd7975677f81ae1a2c2070

**Comment by KamiLordus** [comment]
@spleen1981 I checked the 3D rendering from Longest Journey and it works great. Thanks a lot.
